lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -131,7 +131,7 @@ Flow firrtl::foldFlow(Value val, Flow accumulatedFlow) {
       .Case<SubindexOp, SubaccessOp>(
           [&](auto op) { return foldFlow(op.getInput(), accumulatedFlow); })
       // Registers, Wires, and behavioral memory ports are always Duplex.
-      .Case<RegOp, RegResetOp, WireOp, MemoryPortOp>(
+      .Case<RegOp, RegResetOp, WireOp, MemoryPortOp, BitsPrimOp>(
           [](auto) { return Flow::Duplex; })
```

Comment:
I think this should have the same flow as the parent op, e.g. whatever it is the bits of.

Reply:
I think there is actually a problem here. For example, currently with `bits` getting the same flow as the parent op this FIRRTL has an error when it shouldn't:

```
input a : UInt<8>
output b : UInt<8>
output c : UInt<1>

b <= a
c <= b[0]
```

This should be allowed because even though `b` is a sink,`b[0]` is a passive type so it should be allowed on the RHS. I'm not sure there is any place in the codebase that does this check. Instead it seems like there is a check that allows sinks on the RHS if they are ports, but in this case `b` is a port but `b[0]` is not detected as a port (should it be?).

Not sure the best way to resolve this. I could try to update the flow checker to allow sinks used as sources if they are `bits` of ports, I could switch this back to duplex, or perhaps there is some other solution.

Reply:
I think I have resolved this by handling `bits` properly in `getDeclarationKind`.